kolibri/core/assets/src/api-resources/contentNode.js

```diff
@@ -34,4 +37,51 @@ export default new Resource({
   fetchNextSteps(getParams) {
     return this.fetchDetailCollection('next_steps', Store.getters.currentUserId, getParams);
   },
+  cache: {},
+  fetchModel({ id }) {
```

Comment:
Could you add some comments explaining why you've overridden the caching mechanism specifically for this child of `Resource`? It might help in case there is need for debugging here in the future to know _why_ it was done this way instead of using the default implementation in `Resource`.

Reply:
Can do - also, any thoughts on this implementation (it's essentially a simplified form of the existing implementation, but removing support for dynamic fields loading, and allowing for easier sharing of data for caching across endpoints)?

Reply:
After another pass - one thing I'm not seeing is where the cache is actually going to be read from.

The default implementation of `Resource.fetchCollection` calls  `Resource.getCollection` which does all of the cache checking business.

These implementations of `ContentNodeResource.fetch(Collection | Tree | MoreTree)` never reach into the cache before hitting the API.

Otherwise, this is certainly a far more easily grokked set of functions than all of the cache internals of `Resource`.

Reply:
`fetchModel` and `fetchCollection` are the most commonly used entrypoints to the fetching behaviour. One big change here is that this is no longer trying to do any caching of collection fetching behaviour - instead relying on browser cache to prevent API requests hitting the server in cases where the cache headers are set.

The only thing this really does is to make sure if we do `fetchModel` after a `fetchCollection` that contained that specific model, we don't do another request to the server just to get that one model that we already have all the data for.

So, yeah, in many ways the cache is doing less - and I think it's appropriate for it to do so, especially when combined with good caching headers on the backend.

Reply:
Related to the comment I left about payload sizes, I also noted that there is not `content-encoding` header on the responses - but the client is ready to accept. Are we not gzipping our responses?

Reply:
Not for API responses generally, no - we did have it enabled in CherryPy prior to 0.12.x and it was a source of considerable slowdown.